src/measure_theory/integration.lean

```diff
@@ -519,6 +587,95 @@ end
 
 end measure
 
+section fin_vol_supp
+
+variables [measure_space α] [has_zero β] [has_zero γ]
+
+open finset ennreal
+
+protected def fin_vol_supp (f : α →ₛ β) : Prop := ∀b ≠ 0, volume (f ⁻¹' {b}) < ⊤
+
+lemma fin_vol_supp_map {f : α →ₛ β} {g : β → γ} (hf : f.fin_vol_supp) (hg : g 0 = 0) :
+  (f.map g).fin_vol_supp :=
+begin
+  assume c hc,
+  simp only [map_preimage, volume_bUnion_preimage],
+  apply sum_lt_top,
+  assume b, simp only [mem_filter, mem_range, mem_singleton_iff, and_imp, exists_imp_distrib],
+  assume a fab gbc,
+  apply hf, assume b0,
+  rw [b0, hg] at gbc, rw gbc at hc,
```

Comment:
put one statement per line (unless you are proving a full goal, in which case the proof should be enclosed in braces)